00-test-enhancement.md

```diff
@@ -0,0 +1,67 @@
+# Proposal: Test Enhancement
+
+## Background
+
+Obviously, testing is an important part of the project, but our sdks are not tested well which make them don't work as expect from time to time.
+
+Now, we have migrated all build steps from travisCI to Github Actions, and built our own github action runners. We can redesign and set up our tests logic.
+
+## Proposal
+
+So I propose to do following test enhancement:
+
+### Unit Tests
+
+During unit tests, SDK SHOULD NOT: 
+
+- send any HTTP requests;
+- use any production values, valid access keys and secret keys, production host and zone names, etc.
+
+Tests SHOULD be executed at every commit.
+
+SDK should implement following unit test cases:
+
+#### Build Request
+
+> Following <endpoint> means endpoint without protocol and port.
+
+- `ListBuckets` SHOULD send request `GET /` with host `<endpoint>`
+- `ListObjects` in virtual host style SHOULD send request `GET /` with host `<bucket-name>.<endpoint>`
+- `ListObjects` in path style SHOULD send request `GET /<bucket-name>` with host `<endpoint>`
+- `GetObject` in virtual host style SHOULD send request `GET /<object-key>` with host `<bucket-name>.<endpoint>`
+- `GetObject` in path style SHOULD send request `GET /<bucket-name>/<object-key>` with host `<endpoint>`
+- `PutObject-Copy` in virtual host style SHOULD send request `PUT /<object-key> with host `<bucket-name>.<endpoint>` and header `x-qs-copy-source: <copy-source>`
+
```

Comment:
Should we add all actions here step by step?

Reply:
SDK are not required to implement the whole RFC in the same PR, maintainer can decide their own pace.